tests/7zip_compatiblity_test.rs

```diff
@@ -0,0 +1,115 @@
+#![cfg(feature = "test-7zip")]
+//! This test compresses some random data with deflate64 using p7zip `7z` command and check decompression
+
+use std::ffi::OsString;
+use std::fs::File;
+use deflate64::Deflate64Decoder;
+use std::io::{Cursor, Read, Write};
+use std::mem::size_of;
+use std::process::{Command, Stdio};
+use proptest::proptest;
+use tempfile::TempDir;
+
+const TEST_FILE_NAME: &'static str = "test.file";
+const TEST_ZIP_NAME: &'static str = "test.zip";
+
+#[repr(packed)]
+struct ZipLocalFileHeader {
+    signature: [u8; 4],
+    version: [u8; 2],
+    flags: [u8; 2],
+    compression_method: [u8; 2],
+    last_mod_time: [u8; 2],
+    last_mod_date: [u8; 2],
+    uncompressed_crc32: [u8; 4],
+    compressed_size: [u8; 4],
+    uncompressed_size: [u8; 4],
+    file_name_len: [u8; 2],
+    extra_field_len: [u8; 2],
+}
+
+impl ZipLocalFileHeader {
+    const SIGNATURE: [u8; 4] = [0x50, 0x4B, 0x03, 0x04];
+
+    fn zero() -> Self {
+        unsafe { std::mem::zeroed() }
+    }
+
+    fn as_mut_bytes(&mut self) -> &mut [u8] {
+        unsafe {
+            std::slice::from_raw_parts_mut(
+                self as *mut Self as *mut u8,
+                size_of::<Self>()
+            )
+        }
+    }
+}
+
+fn compress_with_7zip(data: &[u8]) -> Vec<u8> {
+    let mut temp_dir = TempDir::new().unwrap();
+
+    // write data to test file
+    File::create(temp_dir.path().join(TEST_FILE_NAME)).unwrap()
+        .write_all(data).unwrap();
+
+    let seven_zip = std::env::var_os("SEVEN_ZIP_PATH").unwrap_or_else(|| OsString::from("7z"));
+
+    let mut seven_zip_process = Command::new(seven_zip)
+        .arg("a")
+        .arg("-mm=Deflate64")
+        .arg(TEST_ZIP_NAME)
+        .arg(TEST_FILE_NAME)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .stdin(Stdio::null())
+        .current_dir(temp_dir.path())
+        .spawn()
+        .unwrap();
+
+    if !seven_zip_process.wait().unwrap().success() {
+        let mut stdout = String::new();
+        let mut stderr = String::new();
+        seven_zip_process.stdout.unwrap().read_to_string(&mut stdout).unwrap();
+        seven_zip_process.stderr.unwrap().read_to_string(&mut stderr).unwrap();
+        panic!("7zip failure.\nstdout:\n{stdout}\n\nstderr:\n{stderr}");
+    }
+
+    // parse zip file
+    let mut zip_file = File::open(temp_dir.path().join(TEST_ZIP_NAME)).unwrap();
+    let mut header = ZipLocalFileHeader::zero();
+    zip_file.read_exact(header.as_mut_bytes()).unwrap();
+    assert_eq!(header.signature, ZipLocalFileHeader::SIGNATURE);
+    assert_eq!(u16::from_le_bytes(header.flags), 0);
+    assert_eq!(u16::from_le_bytes(header.compression_method), 9);
+    let compressed_size = u32::from_le_bytes(header.compressed_size);
```

Comment:
P.S. You can consider write a [`serde::Deserializer`](https://serde.rs/impl-deserializer.html), then you won't have to write these repeated `u*::from_le_bytes`.

Reply:
Could you let know me how can I use serde to deserialize binary data? I don't know how to use serde in with raw binary data. sorry.

Reply:
You just need to implement [`serde::Deserializer`](https://docs.rs/serde/1.0.183/serde/de/trait.Deserializer.html) according to the [guide](https://serde.rs/impl-deserializer.html), by implementing the deserializing for integers, bools, etc.

You can have a look at [`ssh_format`](https://github.com/openssh-rust/ssh_format), a crate I wrote to deserialize ssh protocol, which is binary, using serde.

`serde` is very flexible and binary formats such as bincode, MessagePack support it, so it's not going to be very hard if you have something to follow.

Reply:
P.S. Whether to implement `serde::Deserializer` depends on how many variables would you need to parse.
The more there are, the more beneficial to write a `serde::Deserializer`.

Reply:
I just read your response that it only need to parse zip entry once, so you probably don't need it.